### ISSUE TITLE
HTTP Error Again

### ISSUE THREAD

**Issue by Zhklu2003** [ShouWenWang-Lab/cospar#11]
Hi Dr. Wang,

I was trying to download the dataset by

import cospar as cs
adata_orig = cs.datasets.hematopoiesis()

But a HTTP error occurs since the url "https://kleintools.hms.harvard.edu/tools/downloads/cospar/LARRY_adata_preprocessed.h5ad" is now inaccessible, 

And also there are some issues about gene_expression_dynamics function.

Could you check it? Thanks!

**Comment by ShouWenWang** [comment]
That is true. Please use the latest CoSpar in our github. This will work.
git clone https://github.com/ShouWenWang-Lab/cospar
python setup.py develop

--
Shou-Wen Wang, Ph.D.
Principal Investigator
School of Life Sciences | School of Science
Westlake University
Shilongshan ST #18, Xihu, Hangzhou, Zhejiang
https://www.shouwenwang-lab.com/



From: Zhklu2003 ***@***.***>
Date: Thursday, June 27, 2024 at 1:44 PM
To: ShouWenWang-Lab/cospar ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [ShouWenWang-Lab/cospar] HTTP Error Again (Issue #11)

Hi Dr. Wang,

I was trying to download the dataset by

import cospar as cs
adata_orig = cs.datasets.hematopoiesis()

But a HTTP error occurs since the url "https://kleintools.hms.harvard.edu/tools/downloads/cospar/LARRY_adata_preprocessed.h5ad" is now inaccessible,

And also there are some issues about gene_expression_dynamics function.

Could you check it? Thanks!

**Comment by Zhklu2003** [comment]
Thanks for help!
Also, I have met a problem. When I run the code below

import cospar as cs
adata_orig = cs.datasets.hematopoiesis_subsampled()
adata = cs.tmap.infer_Tmap_from_multitime_clones(
    adata_orig,
    clonal_time_points=["2", "4"],
    later_time_point="6",
    smooth_array=[20, 15, 10, 5],
    sparsity_threshold=0.1,
    intraclone_threshold=0.2,
    max_iter_N=10,
    epsilon_converge=0.01,
)
cs.tl.progenitor(
    adata,
    selected_fates=["Neutrophil","Monocyte"],
    source="transition_map",
    map_backward=True,
    bias_threshold_A=0.5,
    bias_threshold_B=0.5,
    sum_fate_prob_thresh=0.2,
    avoid_target_states=True,
)
gene_name_list = ["Gata1", "Mpo", "Elane", "S100a8"]
selected_fate = "Neutrophil"
cs.pl.gene_expression_dynamics(
    adata,
    selected_fate,
    gene_name_list,
    traj_threshold=0.2,
    invert_PseudoTime=False,
    compute_new=True,
    gene_exp_percentile=99,
    n_neighbors=8,
    plot_raw_data=False,
)
gene_name_list = ["Gata1", "Mpo", "Elane", "S100a8"]
selected_fate = "Neutrophil"
cs.pl.gene_expression_dynamics(
    adata,
    selected_fate,
    gene_name_list,
    traj_threshold=0.2,
    invert_PseudoTime=False,
    compute_new=True,
    gene_exp_percentile=99,
    n_neighbors=8,
    plot_raw_data=False,
)

I will meet a type error TypeError: asarray() got an unexpected keyword argument 'copy'.

Could you give me some advice? Thanks.

**Comment by Zhklu2003** [comment]
And I am wondering if you have replication code online for this CoSpar paper? Some of figures and experiments cannot be found in https://cospar.readthedocs.io/.

Thanks!
